controllers/grid/SwordDepositPointsGridHandler.inc.php

```diff
@@ -0,0 +1,191 @@
+				'add_item'
+			)
+		);
+		
+		// Columns
+		$cellProvider = new SwordDepositPointsGridCellProvider();
+		$this->addColumn(new GridColumn(
+			'name',
+			'plugins.generic.sword.depositPoints.name',
+			null,
+			'controllers/grid/gridCell.tpl',
+			$cellProvider
+		));
+		$this->addColumn(new GridColumn(
+			'url',
+			'plugins.importexport.sword.depositUrl',
+			null,
+			'controllers/grid/gridCell.tpl',
+			$cellProvider
+		));
+		$this->addColumn(new GridColumn(
+			'type',
+			'common.type',
+			null,
+			'controllers/grid/gridCell.tpl',
+			$cellProvider
+		));
+	}
+	
+	/**
+	 * @copydoc Gridhandler::getRowInstance()
+	 */
+	public function getRowInstance() {
+		return new SwordDepositPointsGridRow();
+	}
+
+	/**
+	 * Display the grid's containing page.
+	 * @param $args array
+	 * @param $request PKPRequest
+	 */
+	public function index($args, $request) {
+		$templateMgr = TemplateManager::getManager($request);
+		$dispatcher = $request->getDispatcher();
+		return $templateMgr->fetchAjax(
+			'swordDepositPointsGridContainer',
+			$dispatcher->url(
+				$request, ROUTE_COMPONENT, null,
+				'plugins.generic.sword.controllers.grid.SwordDepositPointsGridHandler',
+				'fetchGrid'
+			)
+		);
+	}
+
+	/**
+	 * An action to add a new deposit point
+	 * @param $args array Arguments to the request
+	 * @param $request PKPRequest Request object
+	 */
+	public function addDepositPoint($args, $request) {
+		return $this->editDepositPoint($args, $request);
+	}
+
+	/**
+	 * An action to edit a deposit point
+	 * @param $args array Arguments to the request
+	 * @param $request PKPRequest Request object
+	 * @return string Serialized JSON object
+	 */
+	public function editDepositPoint($args, $request) {
+		$depositPointId = $request->getUserVar('depositPointId');
+		$context = $request->getContext();
+		$this->setupTemplate($request);
+		self::$plugin->import('controllers.grid.form.SwordDepositPointForm');
+		$swordDepositPointForm = new SwordDepositPointForm(self::$plugin, $context->getId(), $depositPointId);
+		$swordDepositPointForm->initData();
+		return new JSONMessage(true, $swordDepositPointForm->fetch($request));
+	}
+
+	/**
+	 * Update deposit point
+	 * @param $args array
+	 * @param $request PKPRequest
+	 * @return string Serialized JSON object
+	 */
+	public function updateDepositPoint($args, $request) {
+		$depositPointId = $request->getUserVar('depositPointId');
+		$context = $request->getContext();
+		$this->setupTemplate($request);
+
+		self::$plugin->import('controllers.grid.form.SwordDepositPointForm');
+		$swordDepositPointForm = new SwordDepositPointForm(self::$plugin, $context->getId(), $depositPointId);
+		$swordDepositPointForm->readInputData($request);
+
+		if ($swordDepositPointForm->validate()) {
+			$swordDepositPointForm->execute();
+			return DAO::getDataChangedEvent();
+		} else {
+			// Present any errors
+			$json = new JSONMessage(true, $swordDepositPointForm->fetch($request));
+			return $json->getString();
```

Comment:
It's possible to return a `JSONMessage` directly from a handler; no need to manually convert it to a string.

Reply:
updated.